src/clients/crux-wallet-client.ts

```diff
@@ -0,0 +1,210 @@
+import { CruxDomainId, CruxId, InputIDComponents } from "../packages/identity-utils";
+import { InMemStorage } from "../packages/inmem-storage";
+import { StorageService } from "../packages/storage";
+import { cloneValue } from "../packages/utils";
+
+export const throwCruxClientError = (target: any, prop: any, descriptor?: { value?: any; }): any => {
+    let fn: any;
+    let patchedFn: any;
+    if (descriptor) {
+        fn = descriptor.value;
+    }
+    return {
+        configurable: true,
+        enumerable: true,
+        get() {
+            if (!patchedFn) {
+                patchedFn = async (...params: any[]) => {
+                    try {
+                        return await fn.call(this, ...params);
+                    } catch (error) {
+                        throw CruxClientError.fromError(error);
+                    }
+                };
+            }
+            return patchedFn;
+        },
+        set(newFn: any) {
+            patchedFn = undefined;
+            fn = newFn;
+        },
+    };
+};
+export interface ICruxWalletClientOptions {
+    privateKey?: string;
+    blockstackInfrastructure?: ICruxBlockstackInfrastructure;
+    cacheStorage?: StorageService;
+    walletClientName: string;
+}
+
+export class CruxWalletClient {
+    public walletClientName: string;
+    private cruxBlockstackInfrastructure: ICruxBlockstackInfrastructure;
+    private _initPromise: Promise<void>;
+    private _cruxUser?: CruxUser;
+    private cruxDomain?: CruxDomain;
+    private _cruxUserRepository: ICruxUserRepository;
+    private _cruxAssetTranslator?: CruxAssetTranslator;
+    private _keyManager?: IKeyManager;
+    private resolvedClientAssetMapping?: IResolvedClientAssetMap;
+    private cacheStorage?: StorageService;
+
+    constructor(options: ICruxWalletClientOptions) {
+        setCacheStorage(options.cacheStorage || new InMemStorage());
+        this.cruxBlockstackInfrastructure = options.blockstackInfrastructure || CruxSpec.blockstack.infrastructure;
+        this.walletClientName = options.walletClientName;
+        this._cruxUserRepository = new BlockstackCruxUserRepository({cacheStorage: this.cacheStorage, blockstackInfrastructure: this.cruxBlockstackInfrastructure});
+        this._initPromise = this._init(options);
+    }
+
+    @throwCruxClientError
+    public getCruxIDState = async (): Promise<ICruxIDState> => {
+        await this._initPromise;
+        if (!this._cruxUser) {
+            if (this._keyManager) {
+                return {
+                    cruxID: null,
+                    status: {
+                        status: "NONE",
+                        statusDetail: "",
+                    },
+                };
+            } else {
+                throw ErrorHelper.getPackageError(null, PackageErrorCode.PrivateKeyRequired);
+            }
+        }
+        return {
+            cruxID: this._cruxUser.cruxID.toString(),
+            status : this._cruxUser.registrationStatus,
```

Comment:
This will always return the same status along the lifecycle of the client. We should be refreshing the in-memory cruxUser every time status method is called.

Reply:
Same should be adopted in CruxOnBoardingClient also.

Reply:
Fixed and updated